### PR TITLE
Try to make distance-metrics section clearer

### PR DESCRIPTION
**Pull request by jobergum**
An attempt at making the distance-metric section clearer. I hope you can review this carefully. 

Main points

- Some attempt to use innerproduct distance-metric without normalizing vectors to unit vectors. This breaks metric space properties. 
- Some attempt to transform maximum inner product models (MIP) to unit vectors just by a post-processing step, a transformation that does not preserve the order. 

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Review by arnej27959 — APPROVED** [review]
looks good, but if they didn't read the doc, does it help adding more words?

**Comment by jobergum** [comment]
>looks good, but if they didn't read the doc, does it help adding more words?
Maybe not :)

File: en/reference/schema-reference.html [hunk 1]
```diff
@@ -1580,8 +1580,47 @@ <h2 id="distance-metric">distance-metric</h2>
...
+<ul>
+  <li>Distance is symmetric: <code>distance(q,d) = distance(d,q)</code></li> 
+  <li>Distance is larger than 0 and the distance from any point to itself is zero. <code>distance(q,q) = 0</code></li> 
+  <li>Distance obeys the <a href="https://en.wikipedia.org/wiki/Triangle_inequality">triangle inequality</a>:
+ <code>distance(q,d) &gt;= distance(q,r) + distance(r,d)</code>  
+</ul>
+<p>The distance metric used must match the distance metric used during metric representation learning (model). 
+If you are using an "off-the-shelf" model to vectorize your data, please ensure that the distance metric matches
+the distance metric suggested used with the model. Different type
+of vectorization models uses different type of distance metrics.</p>
```
Comment (geirst) [inline en/reference/schema-reference.html:1598]:
```suggestion
of vectorization models use different type of distance metrics.</p>
```

File: en/reference/schema-reference.html [hunk 2]
```diff
@@ -1580,8 +1580,47 @@ <h2 id="distance-metric">distance-metric</h2>
...
+
+{% include note.html content="
+Some vectorization models, including matrix factorization, uses \"maximum inner product\" (MIP),
```
Comment (geirst) [inline en/reference/schema-reference.html:1601]:
```suggestion
Some vectorization models, including matrix factorization, use \"maximum inner product\" (MIP),
```

File: en/reference/schema-reference.html [hunk 3]
```diff
@@ -1580,8 +1580,47 @@ <h2 id="distance-metric">distance-metric</h2>
...
+with vectors that aren't normalized to unit vectors. These models uses both direction and magnitude.
```
Comment (geirst) [inline en/reference/schema-reference.html:1602]:
```suggestion
with vectors that aren't normalized to unit vectors. These models use both direction and magnitude. 
```

File: en/reference/schema-reference.html [hunk 4]
```diff
@@ -1580,8 +1580,47 @@ <h2 id="distance-metric">distance-metric</h2>
...
+{% include note.html content="
+Some vectorization models, including matrix factorization, uses \"maximum inner product\" (MIP),
+with vectors that aren't normalized to unit vectors. These models uses both direction and magnitude. 
+<strong>MIP</strong> cannot directly be represented using nearest neighbor search distance metrics 
+as it does not obey the metric space properties (triangle inequality). 
+Post processing vectors from a MIP similarity model to
+unit length vectors will cause information loss which will impact the accuracy.  
+"%}
+<p>
+It's possible to do a order preserving transformation of a maximum inner product space
```
Comment (geirst) [inline en/reference/schema-reference.html:1609]:
```suggestion
It's possible to do an order preserving transformation of a maximum inner product space  
```

**Review by geirst — APPROVED** [review]
👍

**Comment by jobergum** [comment]
Thanks @arnej27959 and @geirst !

File: en/reference/schema-reference.html [hunk 5]
```diff
@@ -1580,8 +1580,47 @@ <h2 id="distance-metric">distance-metric</h2>
...
 Only relevant for tensor attribute fields.
+
+Distance metrics used with <a href="../approximate-nn-hnsw.html">HNSW</a> 
+must obey all the defining properties of a <a href="https://en.wikipedia.org/wiki/Metric_space">metric space</a>:</p>
+
+<ul>
+  <li>Distance is symmetric: <code>distance(q,d) = distance(d,q)</code></li> 
+  <li>Distance is larger than 0 and the distance from any point to itself is zero. <code>distance(q,q) = 0</code></li> 
+  <li>Distance obeys the <a href="https://en.wikipedia.org/wiki/Triangle_inequality">triangle inequality</a>:
+ <code>distance(q,d) &gt;= distance(q,r) + distance(r,d)</code>
```
Comment (kkraune) [inline en/reference/schema-reference.html:1593]: distance(q,d) &lt;= distance(q,r) + distance(r,d)

Comment (kkraune) [reply en/reference/schema-reference.html:1593]: I'll put this in a separate PR ...

**Pull request merged**